B3 app manager

orangejenny opened the pull request:
Interdependent with https://github.com/dimagi/Vellum/pull/590

If reviewing commit-by-commit, be aware that a few chunks of this have already been cherry-picked and merged to master (https://github.com/dimagi/commcare-hq/pull/9872, https://github.com/dimagi/commcare-hq/pull/9900, https://github.com/dimagi/commcare-hq/pull/9901, https://github.com/dimagi/commcare-hq/pull/9910).

code buddy @TylerSheffels 
B3 context @biyeun 
app manager context @dannyroberts / @snopoke 
fyi @dimagi/product 

Known issues to be fixed before this is merged:
- [x] B3 version or alternative for `alert_user` (see TODO in detail-screen-config.js)
- [x] Tooltips don't work (e.g., hovering over the pencil icons in the menu)
- [x] Vellum full-screen mode doesn't cover top menu (fixed by https://github.com/dimagi/commcare-hq/pull/9932)
- [x] In the modal to edit graphs, the autocompletes for property names aren't working
- [x] Footer isn't behaving quite right (have to scroll to see the bottom half of it)
- [x] Label under form settings > logic is misplaced when feature flag "Allow disabling Form Builder per form" is on
- [x] App Settings > Custom Properties hasn't been migrated

QA ticket: http://manage.dimagi.com/default.asp?204495

app settings
![1 settings](https://cloud.githubusercontent.com/assets/1486591/12454952/a8655836-bf67-11e5-8594-7be98f69d852.png)

case list
![2 case list](https://cloud.githubusercontent.com/assets/1486591/12454966/af80fa8a-bf67-11e5-951d-7d60940913c2.png)

case list advanced features
![3 case list advanced](https://cloud.githubusercontent.com/assets/1486591/12454975/b754c76e-bf67-11e5-844d-46d7d021d7d6.png)

case detail
![4 case detail](https://cloud.githubusercontent.com/assets/1486591/12454984/bceed78c-bf67-11e5-9c5a-f5f6b0e768a4.png)

form view > case management (basic module)
![5 form case mgmt basic](https://cloud.githubusercontent.com/assets/1486591/12454999/c65dca58-bf67-11e5-93eb-aada20456f01.png)

form view > case management (careplan module)
![6 form case mgmt careplan](https://cloud.githubusercontent.com/assets/1486591/12455001/c667deda-bf67-11e5-85cc-32b15771d8c8.png)

form view > case management (advanced module)
![7 form case mgmt advanced](https://cloud.githubusercontent.com/assets/1486591/12454998/c65c5614-bf67-11e5-9887-dea8de77e354.png)

form view > visit scheduler
![8 form visit scheduler](https://cloud.githubusercontent.com/assets/1486591/12455002/c667de94-bf67-11e5-8f47-4a81b472186c.png)

deploy page
![9 deploy](https://cloud.githubusercontent.com/assets/1486591/12455000/c65ed646-bf67-11e5-8561-7dce3ee4484a.png)

orangejenny committed:
update parent templates, content blocks, and template tags

orangejenny committed:
js libraries

orangejenny committed:
css classes: span* and offset*

orangejenny committed:
css classes: buttons

orangejenny committed:
namespace events

orangejenny committed:
modals

orangejenny committed:
icons

orangejenny committed:
CSS: row-fluid => row

orangejenny committed:
CSS: button sizes

orangejenny committed:
CSS: alerts

orangejenny committed:
CSS: forms

orangejenny committed:
CSS: .list-unstyled

orangejenny committed:
CSS: muted => text-muted

orangejenny committed:
CSS: labels

orangejenny committed:
CSS: accordions

orangejenny committed:
TODOs for classes that have been deprecated without obvious equivalents

orangejenny committed:
create B3 version of bulk_upload.html

orangejenny committed:
fix smart quotes

orangejenny committed:
Convert key_value_mapping.html to B3

orangejenny committed:
TODO to fix sized inputs

orangejenny committed:
Convert value-or-none-ui.html to B3

orangejenny committed:
include knockout

orangejenny committed:
include jquery ui, exclude jqmodal

orangejenny committed:
fix breadcrumbs

orangejenny committed:
clean up import_app page

orangejenny committed:
add btn-default to save button

orangejenny committed:
clean up module settings tab

orangejenny committed:
clean up form settings tab

orangejenny committed:
merge master

orangejenny committed:
use base_section.html template

orangejenny committed:
Use knockout 2 instead of 3

orangejenny committed:
Include multiTypeahead widget on all app manager pages

orangejenny committed:
stop hiding language select in menu

orangejenny committed:
menu: fix pencil placement

orangejenny committed:
.nav-header => .text-hq-nav-header

orangejenny committed:
add divider styles

orangejenny committed:
make COMMCAREHQ.icons B3-compatible

orangejenny committed:
menu fixes

orangejenny committed:
fix breadcrumbs

orangejenny committed:
deploy page: fix starring

orangejenny committed:
deploy page: smarten up form to edit comment

orangejenny committed:
merge master

orangejenny committed:
fix bad blocktrans

orangejenny committed:
styling for deploy modal

orangejenny committed:
support 'scan bar code' modal

orangejenny committed:
hack at menu spacing

orangejenny committed:
straighten up Languages > Language List

orangejenny committed:
fix indenting of modules & forms in menu

orangejenny committed:
Use B3 version of bulk upload in Languages > Bulk Application Translations

orangejenny committed:
fix positioning of deleted app modal

orangejenny committed:
Straighten up Languages > CommCare Translations

orangejenny committed:
Minor style tweaks to care plan module creation modal

orangejenny committed:
straighten up app settings page

orangejenny committed:
drop unused page-level styles

orangejenny committed:
Deprecate password reset widget in favor of a standard modal

orangejenny committed:
float save button back to the right

orangejenny committed:
Stop collapsing sections whenever a setting is changed

orangejenny committed:
make bootstrap-tab-history-custom.js B3 compatible

orangejenny committed:
mark multimedia_uploader.html as B2 (cherry-pick to master)

orangejenny committed:
add B3 version of multimedia_uploader.html and dependent partials (cherry-pick)

orangejenny committed:
move hqm_upload_form.html to B2 directory (cherry-pick)

orangejenny committed:
use B2 hqm_upload_form.html on B3 multimedia_uploader.html (cherry-pick)

orangejenny committed:
Use B3 MM uploader in app manager

orangejenny committed:
Straighten up form settings > display condition and case list > case filter

orangejenny committed:
drop unused styles

orangejenny committed:
Straighten up form > advanced tab

orangejenny committed:
Use legacy version of jquery ui

orangejenny committed:
merge master

orangejenny committed:
upgrade jquery-ui

orangejenny committed:
Straighten up Form View > User Case Management

orangejenny committed:
Straighten up Form View > Case Management

orangejenny committed:
Update help icons

orangejenny committed:
make help bubbles work

orangejenny committed:
Move help bubble processing into base template

orangejenny committed:
Take main.js out of import app page

orangejenny committed:
On settings page, move 'not an allowed value' error and make it more intimidating

orangejenny committed:
make accordion icons change on expand/collapse

orangejenny committed:
Drop some completed TODOs

orangejenny committed:
give help icons width (cherry-pick)

orangejenny committed:
clean up Form View > Logic > EOF nav > form links

orangejenny committed:
fix js tests

orangejenny committed:
Migrate copy application

orangejenny committed:
fix breadcrumb for editing form

orangejenny committed:
position vellum correctly within form edit page

orangejenny committed:
Clean up care plan module settings tab

orangejenny committed:
make reporting module's settings tab use module_filter.html

orangejenny committed:
straighten up report module settings tab

orangejenny committed:
translations for password reset

orangejenny committed:
remove extra space at the top of managed app pages

orangejenny committed:
migrate css for Reports tab

orangejenny committed:
Migrate module view > case management

orangejenny committed:
fix spacing on media's 'Set Icon/Audio Path' feature

orangejenny committed:
fix styling for module view > case management > auto select case

orangejenny committed:
merge master

orangejenny committed:
restyle module view > case mgmt > reg form from case list

orangejenny committed:
merge master

orangejenny committed:
Update defaults calls to be compatible with Underscore 1.5.0+

orangejenny committed:
straighten up case details

orangejenny committed:
use legends on form advanced tab, to match other tabs

orangejenny committed:
visit scheduler tab (module)

orangejenny committed:
Visit Scheduler tab for forms, part 1

orangejenny committed:
finish styling form's Visit Scheduler tab

orangejenny committed:
restyle graphing modal

orangejenny committed:
reposition lang codes

orangejenny committed:
Migrate case list

orangejenny committed:
Fix vertical alignment of media buttons (sticking too far to the right)

orangejenny committed:
tweak style of menu's languages dropdown

orangejenny committed:
Replace case management's double wells with panels

orangejenny committed:
Restyle case management for forms in advanced modules

orangejenny committed:
some more styling for advanced forms

orangejenny committed:
missing trans

orangejenny committed:
merge master

orangejenny committed:
fix nav media modal not showing existing media

orangejenny committed:
get rid of .help-inline and make validation-related fixes

orangejenny committed:
more consistent vertical spacing on managed app pages

orangejenny committed:
get accordion to respond

orangejenny committed:
merge master

orangejenny committed:
advacned forms: always use accordions

orangejenny committed:
advanced forms: restyle actions

orangejenny committed:
advanced forms: fix alignment of case preload/save panels

orangejenny committed:
advanced form: combine panels

orangejenny committed:
mismatched tag

orangejenny committed:
advanced form: fix spacing of legends

orangejenny committed:
merge master

orangejenny committed:
drop some TODOs

dimagimon commented:
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

benrudolph commented:
holy :hankey:! this is monstrous! :raised_hands: :raised_hands: :raised_hands:

snopoke commented:
Indeed a monter! :ocean: :whale2:

czue commented:
wow! amazing to see this happening! :dragon:

biyeun commented:
will take a pass after gym. but :+1: SO HAPPY.

biyeun commented on corehq/apps/app_manager/static/app_manager/json/commcare-app-settings.yaml:
```diff
@@ -146,7 +146,7 @@
 
 - id: auto_gps_capture
-  name: 'Auto Capture Location (all forms)'
+  name: 'Auto Capture Location<br>(all forms)'
```
`<br />`

biyeun commented on corehq/apps/app_manager/templates/app_manager/app_view.html:
```diff
@@ -295,38 +288,43 @@
-            <label class="control-label">{% trans "Password" %}</label>
-            <div class="controls">
-                <input type="password" class="form-control" />
+<<<<<<< HEAD
```
`<<<<<HEAD`

snopoke replied:
:suspect:

dannyroberts commented:
:open_mouth: Incredible! :rainbow:

biyeun commented on corehq/apps/app_manager/templates/app_manager/apps_base.html:
```diff
@@ -125,18 +133,25 @@
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h3 class="modal-title">{% trans 'Oops! This application was deleted.' %}</h3>
```
`h4`

orangejenny replied:
yeah, i did h3 everywhere, it was what the migration guide happened to have. updated migration guide, will fix the PR.

biyeun commented on corehq/apps/app_manager/templates/app_manager/app_view.html:
```diff
@@ -295,38 +288,43 @@
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Set Password" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/odk_install.html:
```diff
@@ -1,28 +1,32 @@
+            <button type="button" class="close" data-dismiss="modal">
+                <span aria-hidden="true">&times;</span>
+            </button>
+            <h3 class="modal-title">
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/case_config.html:
```diff
@@ -3,16 +3,22 @@
+        <div class="modal-content">
+            <div class="modal-header">
+                <a href="#" class="close" data-dismiss="modal">×</a>
+                <h3 class="modal-title">{% trans "Remove Subcase?" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/case_config_advanced.html:
```diff
@@ -3,16 +3,22 @@
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Remove Form Action?" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/download_async_modal.html:
```diff
@@ -20,12 +20,14 @@
+                <button type="button" class="close" data-dismiss="modal">
+                    <span aria-hidden="true">&times;</span>
+                </button>
+                <h3 class="modal-title">{% trans "Downloading Your File" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html:
```diff
@@ -2,104 +2,118 @@
+                        <div class="modal-content">
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                                <h3 class="modal-title">{% trans "Upload XForm" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html:
```diff
@@ -2,104 +2,118 @@
+                            {% csrf_token %}
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                                <h3 class="modal-title source-readonly">{% trans "XML Source" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/form_tab_advanced.html:
```diff
@@ -2,104 +2,118 @@
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                                <h3 class="modal-title source-readonly">{% trans "XML Source" %}</h3>
+                                <h3 class="modal-title source-edit">{% trans "Edit XML Source" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/graph_configuration_modal.html:
```diff
@@ -1,199 +1,218 @@
+                    <button type="button" class="close" data-dismiss="modal">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                    <h3 class="modal-title" data-bind="text: 'Editing ' + graphDisplayName()"></h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html:
```diff
@@ -1,73 +1,81 @@
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                <h3 class="modal-title">{% trans 'Upload Icon Image' %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/nav_menu_media_modals.html:
```diff
@@ -1,73 +1,81 @@
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                <h3 class="modal-title">{% trans 'Upload Menu Audio' %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/new_careplan_module.html:
```diff
@@ -1,34 +1,39 @@
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                    <h3 class="modal-title">{% trans "New Care Plan Module" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/partials/releases_deploy_modal.html:
```diff
@@ -1,239 +1,262 @@
+            <button type="button" class="close" data-dismiss="modal">
+                <span aria-hidden="true">&times;</span>
+            </button>
+            <h3 class="modal-title">{% trans "Deploying version" %} <span data-bind="text: version"></span></h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/releases.html:
```diff
@@ -308,31 +308,43 @@ <h6 data-bind="if: !built_with.signed()">{% trans "Unsigned Jar" %}</h6>
+                        <button type="button" class="close" data-dismiss="modal">
+                            <span aria-hidden="true">&times;</span>
+                        </button>
+                        <h3 class="modal-title">{% trans "Delete Version?" %}</h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/releases.html:
```diff
@@ -308,31 +308,43 @@ <h6 data-bind="if: !built_with.signed()">{% trans "Unsigned Jar" %}</h6>
+                        <button type="button" class="close" data-dismiss="modal">
+                            <span aria-hidden="true">&times;</span>
+                        </button>
+                        <h3 class="modal-title">{% trans "Revert to Version?" %}</h3>
```
`h4`

biyeun commented on corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html:
```diff
@@ -0,0 +1,99 @@
+                    <button type="button" class="close" data-dismiss="modal">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                    <h3 class="modal-title" data-bind="text: $data.modalTitle"></h3>
```
`h4`

biyeun commented on corehq/apps/app_manager/templates/app_manager/import_app.html:
```diff
@@ -32,29 +31,42 @@
-        <a href="#" class="historyBack">No, take me back.</a>
-        <button class="btn btn-primary" type="submit">{% trans 'Yes, import application' %}</button>
+            <div class="form-group">
+                <label class="col-lg-2 control-label">Name</label>
```
what about `md` and `sm` views? most of our users fall in that category...

biyeun replied:
don't forget about `col-sm` and `col-md` too

biyeun commented on corehq/apps/app_manager/templates/app_manager/module_view.html:
```diff
@@ -210,6 +206,7 @@
     {% endfor %}
     {% endblock %}{# endblock tab-headers #}
 </ul>
+<br>
```
`<br />`

biyeun commented on corehq/apps/app_manager/templates/app_manager/module_view.html:
```diff
@@ -217,20 +214,20 @@
-                    <div class="controls">
-                        {% input_trans module.name langs %}
+                <div class="form-group">
+                    <label class="col-lg-2 control-label">{% trans "Module Name" %}</label>
```
don't forget about `col-sm` and `col-md` too

biyeun replied:
don't forget about `col-sm` and `col-md` too

biyeun commented on corehq/apps/app_manager/templates/app_manager/module_view_advanced.html:
```diff
@@ -48,16 +48,16 @@
-    <div class="control-group">
-        <label class="control-label">
+    <div class="form-group">
+        <label class="control-label col-lg-2">
```
don't forget about `col-sm` and `col-md` too

biyeun commented on corehq/apps/app_manager/templates/app_manager/module_view_careplan.html:
```diff
@@ -20,18 +20,19 @@
-            <select name="parent_module" data-bind="optstr: moduleOptions, value: moduleId"></select>
+    <div id="detail-screen-parent" class="form-group">
+        <label class="control-label col-lg-2">{% trans "Attached to" %}</label>
+        <div class="col-lg-4">
```
don't forget about `col-sm` and `col-md` too

biyeun commented:
Started leaving a few comments, but just want to make sure that there aren't any solo `col-lg` styles without `col-sm` and `col-md` because most of our users don't have large screens!

Also, as a todo for the future, (@dannyroberts) would be good to component-alize (knockout 3) modals. For the other b3 pages i've been using the following in the `modal-header` for consistency (note `h4` vs `h3` semantic usage).:
```<div class="modal-header">
    <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
    <h4 class="modal-title">{{TITLE}}</h4>
</div>
```
If this were a component life would be so much easier *sigh*. :crying_cat_face:

biyeun commented on corehq/apps/hqwebapp/static/hqwebapp/js/ui-element.js:
```diff
@@ -93,9 +93,9 @@ var uiElement;
                 var langcode_button = langcodeTag.button_tag(
-                    $('<a href="#" class="btn btn-inverse btn-mini lang-text"'
-                      + ' style="color:#ffffff; text-decoration: none;" />'),
+                    $('<a href="#" class="btn btn-info btn-xs btn-mini lang-text" style="position: absolute; top: 6px; right: 6px;" />'),
```
guessing this is still b2 supported? `btn-mini`

orangejenny replied:
I didn't check where all ui element is used, but looking now, does look like it's mostly app manager, with one reference in `corehq/apps/groups/templates/groups/group_members.html` (not sure if that's B2 or B3). I'm inclined to leave it B2-compatible until we're all switched over and then do a final sweep for B2 classes. There are other js files that'll need that kind of cleanup anyway (the multimedia popup comes to mind).

biyeun commented on corehq/apps/style/templates/style/bootstrap3/partials/key_value_mapping.html:
```diff
@@ -0,0 +1,99 @@
+            </div>
+        </fieldset>
+
+        <div class="control-group">
```
`form-group`

biyeun commented:
pt 1 review complete

orangejenny commented:
@biyeun  thanks for going through all this.

> just want to make sure that there aren't any solo col-lg styles without col-sm and col-md because most of our users don't have large screens!

Ah, there's a ton of these, I'll take a pass at fixing them. What's our best practice? Including `sm`, `md`, and `lg` everywhere? (Talking about places where we don't care about varying the proportions on different screen sizes, so it'd be `col-lg-2 col-md-2 col-sm-2`.)

> would be good to component-alize (knockout 3) modals

Strongly agree! Doing this migration got me excited about looking into knockout's components. Upgrading to knockout 3 should be a priority after the B3 migration is over (for the hopefully few areas where knockout upgrade wasn't done along with B3 upgrade).

biyeun commented:
@orangejenny then the best option is to default to the smaller screen so instead of `lg` use `sm`

orangejenny commented:
Cool, will do.

dannyroberts committed:
Merge remote-tracking branch 'origin/b3-app-manager' into b3-app-manager+validate-xpath

dannyroberts committed:
Merge branch 'validate-xpath' into b3-app-manager+validate-xpath

Conflicts:
	corehq/apps/app_manager/templates/app_manager/form_view_base.html
	corehq/apps/app_manager/templates/app_manager/module_view.html
	corehq/apps/app_manager/templates/app_manager/module_view_report.html
	corehq/apps/style/templates/style/bootstrap3/partials/value-or-none-ui.html

snopoke commented on corehq/apps/app_manager/static/app_manager/js/case-config-ui-advanced.js:
```diff
@@ -152,18 +152,16 @@ var AdvancedCase = (function () {
-                if (!type || type === 'load') {
-                    $('#case-load-accordion').accordion("destroy").accordion(options);
-                }
-            });
```
I'd be quite happy to get rid of these accordion's completely

orangejenny replied:
It's on my todo list :)  Hoping to squeeze it into this PR.

snopoke commented on corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js:
```diff
@@ -494,7 +497,9 @@ var DetailScreenConfig = (function () {
-            _.defaults(this.original, tabDefaults);
+            _.each(_.keys(tabDefaults), function(key) {
+                that.original[key] = that.original[key] || tabDefaults[key];
+            });
```
Why the change from defaults

orangejenny replied:
Underscore 1.5 updated defaults to only override undefined values. Previously it had been all falsy values, which is what this code needs (it's largely overwriting nulls, not undefined).

snopoke commented on corehq/apps/app_manager/static/app_manager/js/case-knockout-bindings.js:
```diff
@@ -101,9 +101,7 @@ ko.bindingHandlers.casePropertyTypeahead = {
         var options = valueAccessor() || {};
-        setTimeout(function() {
-            $(element).accordion(options);
-        }, 0);
```
I see a few accordion related changes but assume you've tested them

orangejenny replied:
I didn't thoroughly dig through why all of them were necessary (in particular, swapping `load` and `save` in `case-config-ui-advanced.js` felt weird) but yeah, I did test the accordion.

snopoke commented on corehq/apps/app_manager/templates/app_manager/form_view_base.html:
```diff
@@ -339,30 +297,17 @@
+                        data from a form to the case, which will store the data locally on the phone to use later.
+                    </div>
+                    <div class="casexml" id="casexml_home">
+                    {% if app.application_version != '1.0' %}
```
Removed support for `1.0` apps. Getting closer to :skull:

orangejenny replied:
Awesome - can I remove all 1.0-related code in app manager, then?

snopoke replied:
Not quite yet, but hopefully in the next month or so.

snopoke commented on corehq/apps/app_manager/templates/app_manager/languages.html:
```diff
@@ -34,26 +35,25 @@
                         <p class="help-block">
                             {% blocktrans %}
-                            For Android refer to
+                            For Android, refer to
```
Even grammar corrections :angel:

orangejenny committed:
that's just embarrassing

dimagimon commented:
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

dannyroberts committed:
Merge remote-tracking branch 'origin/b3-app-manager+validate-xpath' into b3-app-manager

dimagimon commented:
There are 8 errors:

* corehq/apps/app_manager/templatetags/xforms_extras.py, line 7 - E501 line too long (131 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 51 - E501 line too long (164 > 115 characters)
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 58 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 59 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/templatetags/xforms_extras.py, line 62 - E126 continuation line over-indented for hanging indent
* corehq/apps/app_manager/views/view_generic.py, line 42 - E302 expected 2 blank lines, found 1
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 244 - Bad line breaking before '&&'.
* corehq/apps/app_manager/static/app_manager/js/commcaresettings.js, line 245 - Bad line breaking before '&&'.

orangejenny committed:
dimagimon

orangejenny committed:
Merge branch 'b3-app-manager' of github.com:dimagi/commcare-hq into b3-app-manager

dimagimon commented on corehq/apps/app_manager/templatetags/xforms_extras.py:
```diff
@@ -45,17 +48,25 @@ def html_name(name):
+                        "lang": lang,
+                        "extra_class": " langcode-input",
+                        "extra_style": "position: absolute; top: 6px; right: 15px"
+                    }
```
E123 closing bracket does not match indentation of opening bracket's line

orangejenny committed:
PR feedback

orangejenny committed:
col-lg-* => col-sm-*

orangejenny committed:
Merge branch 'master' into b3-app-manager

orangejenny committed:
Fix auto gps checkbox alignment & form settings > logic disabled messgaes

orangejenny committed:
Migrate app settings > custom properties

orangejenny committed:
Copy alert_user functionality over from B2's base.html

orangejenny committed:
Fix sticky footer

Still not sure why the footer was displaying too low for app manager pages but fine for other B3 pages (e.g., exports) but removing the padding seems to fix app manager without hurting exports.

orangejenny committed:
fix autocompletes in graphing modal

orangejenny committed:
Fix display of less errors in console

orangejenny committed:
Make tooltips work again

orangejenny committed:
Reposition tooltips

orangejenny committed:
Fix audio preview not appearing for menu audio

orangejenny committed:
merge master

orangejenny committed:
Merge branch 'master' into b3-app-manager

orangejenny committed:
Make select2 controls on form case management all the same size

orangejenny committed:
missing .default-btn in multiemedia modal

orangejenny committed:
fix compress test

orangejenny commented:
killing some intermediate builds

orangejenny committed:
missed an h3 => h4

orangejenny committed:
can't include detail-screen-config.js until previews & toggles have been defined

orangejenny committed:
more futzing with the footer

orangejenny committed:
Merge master

orangejenny committed:
Merge branch 'master' into b3-app-manager

orangejenny committed:
merge master

orangejenny committed:
Make menu padding changes specific to app manager

emord committed:
Remove jquery-migrate from formdesigner

orangejenny committed:
drop formdesigner conditional from hq-oldstyle-js.html

orangejenny committed:
janky fix for save button not enabling

orangejenny committed:
Better fix for save button not enabling

orangejenny committed:
Fix misplaced languages dropdown text

orangejenny committed:
Drop lingering inline styles from managed_app.html

orangejenny commented:
QA passed, this is good to merge once https://github.com/dimagi/Vellum/pull/590 is merged. Would like to deploy today.

dannyroberts commented:
Woohoo! :star2:

czue commented:
beastly! :bear: :wolf: :tiger:
